core/src/main/scala/org/http4s/internal/parsing/Rfc5234.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.internal.parsing
+
+import cats.parse.{Parser => P, Parser1}
+
+/** Common rules defined in RFC5234 and imported into various HTTP
+  * specs.
+  *
+  * @see https://tools.ietf.org/html/rfc5234#appendix-B.1
+  */
+private[http4s] object Rfc5234 {
+  val alpha: P[Char] =
+    P.charIn(0x41.toChar to 0x5a.toChar).orElse1(P.charIn(0x61.toChar to 0x7a.toChar))
+  val cr: Parser1[Unit] = P.char(0x0d)
```

Comment:
cats-parse returns `Parser1[Unit]` for the constants and `Parser1[Char]` when there is a choice.  I followed that convention here.  Is there a drawback to making these `Parser1[Char]` across the board?

Reply:
No we could do that. I made Parser.string return Unit and then I thought Parser.char should match string.

The downside is void comes up a lot, and then we would need a wrapper to change the type (Void(p1)).

Open an issue on cats-parse and we can discuss more the design space.

Reply:
Also note: `someParser.string` captures the string of the whole parser by not capturing anything, then using the offsets to do one final substring at the end. If you do want the whole string at the end, that is usually not bad at all.

If you literally want Parser.string(str).string that is a waste since it if matches we wouldn't have to allocate: we already have the string, but I guess I can optimize that into a map that returns the original string without a copy.